src/components/resourceSelect.tsx

```diff
@@ -50,7 +45,7 @@ export function ResourceSelect(props: {
 
   const fetchDropdownResource = async (id: IdEither) => {
     try {
-      const [res] = await fetchSingleResource(id);
+      const [res] = !isEmpty(Object.values(id).join('')) && (await fetchSingleResource(id));
```

Comment:
```
IdEither = {
      id: number;
    }
  | {
      externalId: string;
    };
```
1. `IdEither` shouldn't be empty, if it is then it means types aren't correct
2. This is a bit hackish check anyway, could be at least `Object.values(id).filter(Boolean)`

Reply:
without this ` !isEmpty(Object.values(id).join('')) `, every time as you navigate on timeseries tab will be triggered a call with the `id: ''` , I think is coming from the `defaultQuery` is this wanted?

Reply:
got you!